test/customValidators_test.js

```diff
@@ -150,11 +174,149 @@ describe('validateServerVariables()', function () {
+  it('should successfully validate the server variables has only default and example, no enum', async function () {
+    const inputString = `{ 
+      "servers": {
+        "dummy":
+        {
+          "url": "http://localhost:{port}",
+          "description": "The production API server",
+          "protocol": "secure-mqtt",
+          "variables": {
+            "port": {
+              "default": "8883",
+              "examples" : "8883"
+            }
+          }
+        }
+      }
+    }`;
+
+    const parsedInput = JSON.parse(inputString);
+
+    expect(validateServerVariables(parsedInput, inputString, input)).to.equal(
+      true
+    );
+  });
+
+  // server with a variable that has one example and it doesn't match any of provided enum
+  it('should throw error on the server variables has one example and it does not match any of provided enum', async function () {
+    const inputString = `{ 
+      "servers": {
+        "dummy":
+        {
+          "url": "http://localhost:{port}",
+          "description": "The production API server",
+          "protocol": "secure-mqtt",
+          "variables": {
+            "port": {
+              "enum": ["8883", "8884"],
+              "examples" : "8882"
+            }
+          }
+        }
+      }
+    }`;
+
+    const parsedInput = JSON.parse(inputString);
+
+    expect(() =>
+      validateServerVariables(parsedInput, inputString, input)
+    ).to.throw(
+      'Please check your server variables. The example value is not one of the values from the enum'
+    );
+  });
+
+  // server with a variable that has more than one example and only one of them match enum list,
+  // but the rest don't,
+  // so validation should fail with clear information which example is wrong and where in the file is it
+  it('should throw error on the server variables has one example and only one of them match enum list, but the rest do not', async function () {
+    const inputString = `{ 
+      "servers": {
+        "dummy":
+        {
+          "url": "http://localhost:{port}/{basePath}",
+          "description": "The production API server",
+          "protocol": "secure-mqtt",
+          "variables": {
+            "port": {
+              "enum": ["8883", "8884"],
+              "examples" : "8883",
+              "default": "8883"
+            },
+            "basePath": {
+              "enum": ["v1", "v2", "v3"],
+              "examples" : "v4",
+              "default": "v2"
+            }
+          }
+        }
+      }
+    }`;
+
+    const parsedInput = JSON.parse(inputString);
+
+    try {
+      validateServerVariables(parsedInput, inputString, input);
+    } catch (e) {
+      expect(e.type).to.equal(
+        'https://github.com/asyncapi/parser-js/validation-errors'
+      );
+      expect(e.title).to.equal(
+        'Please check your server variables. The example value is not one of the values from the enum'
```

Comment:
how about 
`Check your server variables. The example does not match the enum list`
?

Reply:
agree!